### PR TITLE
Use more ES6 syntaxes in the shopping cart example

### PR DESCRIPTION
**Pull request by ChrisJamesC**
Convert the shopping cart example files to using fat arrows, spread operators and stateless components.

References  #1800

### CODE REVIEW

File: examples/shopping-cart/src/actions/index.js [hunk 1]
```diff
@@ -1,50 +1,38 @@
...
+  if (getState().products.byId[productId].inventory > 0) {
+    dispatch(addToCartUnsafe(productId))
   }
 }
 
-export function checkout(products) {
-  return (dispatch, getState) => {
-    const cart = getState().cart
+export const checkout = (products) => (dispatch, getState) => {
+  const cart = getState().cart
```
Comment (markerikson) [inline examples/shopping-cart/src/actions/index.js]: If you're really going this far, you might as well do `const {cart} = getState()`.

Comment (ChrisJamesC) [inline examples/shopping-cart/src/actions/index.js]: Done. I agree, it was a total overkill before ;)

**Comment by ellbee** [comment]
Thanks!

File: examples/shopping-cart/src/components/Cart.js [hunk 2]
```diff
@@ -1,37 +1,33 @@
...
-        <div>{nodes}</div>
-        <p>Total: &#36;{total}</p>
-        <button onClick={onCheckoutClicked}
-          disabled={hasProducts ? '' : 'disabled'}>
-          Checkout
-        </button>
-      </div>
-    )
-  }
+  return <div>
```
Comment (gaearon) [inline examples/shopping-cart/src/components/Cart.js]: I strongly dislike this style. Can we please do a pass in a separate PR that ensures returned multilne JSX is always wrapped in parens? As in:

```js
return (
  <div>
    ...
  </div>
)
```

and

```js
const Stuff = () => (
  <div>
    ...
  </div>
);
```

File: examples/shopping-cart/src/reducers/products.js [hunk 3]
```diff
@@ -49,10 +49,8 @@ export default combineReducers({
...
 
-export function getProduct(state, id) {
-  return state.byId[id]
-}
+export const getProduct = (state, id) => state.byId[id]
 
-export function getVisibleProducts(state) {
-  return state.visibleIds.map(id => getProduct(state, id))
-}
+export const getVisibleProducts = state => state.visibleIds.map(id =>
```
Comment (gaearon) [inline examples/shopping-cart/src/reducers/products.js]: If the arrow body is multiline please always start the body on the next line.
Please make it like:

```js
export const getVisibleProducts = state =>
  ...
```

or

```js
export const getVisibleProducts = state => (
  ...
)
```

Comment (ChrisJamesC) [inline examples/shopping-cart/src/reducers/products.js]: Thanks for the styling advices, I'll do a pass on all of my PRs to conform with the comments you made.
